Src/Support/Google.Apis.Auth.Tests/OAuth2/TokenRefreshManagerTests.cs

```diff
@@ -317,5 +318,68 @@ public async Task RetriesError_Async()
             Assert.Equal(TokenRefreshManager.RefreshTimeouts.Length, refreshCallCount);
             Assert.Equal(0, Interlocked.Add(ref refreshCompleted, 0));
         }
+
+        [Fact]
+        public async Task UnobservedException()
+        {
+            // An unobserved exception used to happen when the token is soft expired so that
+            // a refresh token task is started but not inmediately observed and it fails.
+            // See https://github.com/googleapis/google-api-dotnet-client/issues/2021
+            string exceptionMessage = "While testing for unobserved exceptions the refresh task failed.";
+            int unobservedCount = 0;
+            TaskScheduler.UnobservedTaskException += (sender, e) =>
+            {
+                if (e.Exception.InnerExceptions.Any(ex => ex.Message == exceptionMessage))
+                {
+                    Interlocked.Increment(ref unobservedCount);
+                    e.SetObserved();
+                }
+            };
+
+            var refreshCompletionSource = new TaskCompletionSource<bool>();
+            var clock = new MockClock(new DateTime(2010, 1, 1, 0, 0, 0, DateTimeKind.Utc));
+            var logger = new NullLogger();
+            var softExpiredToken = new TokenResponse
+            {
+                AccessToken = clock.UtcNow.ToString("O"),
+                ExpiresInSeconds = TokenResponse.TokenRefreshTimeWindowSeconds,
+                IssuedUtc = clock.UtcNow
+            };
+            TokenRefreshManager trm = new TokenRefreshManager(async ct =>
+            {
+                try
+                {
+                    // Lets wait a little so that we don't complete syncronously.
+                    await Task.Delay(TimeSpan.FromMilliseconds(10));
+                    throw new Exception(exceptionMessage);
+                }
+                finally
+                {
+                    refreshCompletionSource.SetResult(true);
+                }
+            }, clock, logger)
+            {
```

Comment:
Possibly put this just as a separate assignment statement? It's a bit weird as an object initializer here after a large anonymous function.
Alternatively, create a private method so we can just have:

```csharp
TokenRefreshManager trm = new TokenRefreshManager(FailToRefresh, clock, logger)
{
    // The initial token is soft expired.
    Token = softExpiredToken
};
```

I don't mind which.